src/kOS/UserIO/TelnetMainServer.cs

```diff
@@ -0,0 +1,449 @@
+        private void SetPermanentListenPermission(bool newValue)
+        {
+            try
+            {
+                PluginConfiguration savedPermissions = PluginConfiguration.CreateForType<TelnetMainServer>();
+                savedPermissions.load();
+                savedPermissions.SetValue("PermanentListenPermission", newValue);
+                savedPermissions.save();
+            }
+            catch (Exception ex)
+            {
+                UnityEngine.Debug.LogError("kOS: Exception Loading TelnetMainServer.xml (maybe the first time you ran and its not there yet): " + ex.Message);
+            }
+            return; // In principle it should never reach here.
+        }
+
+        /// <summary>
+        /// Return the user's permanent ("don't remind me again") status for the
+        /// permission to have the telnet server listen to an address other han
+        /// loopback.  (false = please restrict to only loopback.)  The permission
+        /// will be stored in the kOS settings file and will be presumed to be set
+        /// to false if the setting is missing from the file (as it will be the first
+        /// time the user runs a version of the mod with this code in it.
+        /// </summary>
+        /// <returns>The permission as read from the kOS settings file</returns>
+        private bool GetPermanentRealIPPermission()
+        {
+            try
+            {
+                PluginConfiguration savedPermissions = PluginConfiguration.CreateForType<TelnetMainServer>();
+                savedPermissions.load();
+                return (bool)(savedPermissions["PermanentRealIPPermission"]);
+            }
+            catch (Exception ex)
+            {
+                UnityEngine.Debug.LogError("kOS: Exception Loading TelnetMainServer.xml (maybe the first time you ran and its not there yet): " + ex.Message);
+            }
+            return false; // In principle it should never reach here.
+        }
+
+        /// <summary>
+        /// Change the permanent ("don't remind me again") status of the permission
+        /// in the kOS settings file.
+        /// </summary>
+        private void SetPermanentRealIPPermission(bool newValue)
+        {
+            try
+            {
+                PluginConfiguration savedPermissions = PluginConfiguration.CreateForType<TelnetMainServer>();
+                savedPermissions.load();
+                savedPermissions.SetValue("PermanentRealIPPermission", newValue);
+                savedPermissions.save();
+            }
+            catch (Exception ex)
+            {
+                UnityEngine.Debug.LogError("kOS: Exception Loading TelnetMainServer.xml (maybe the first time you ran and its not there yet): " + ex.Message);
+            }
+            return; // In principle it should never reach here.
+        }
+        
+        public void SetConfigEnable(bool newVal)
+        {
+            bool isLoopback = (bindAddr == IPAddress.Loopback);
```

Comment:
this will always be a reference comparison rather than value. is that what you wanted?